src/runtime/verifier.c

```diff
@@ -276,12 +262,18 @@ static inline int verify_string(const void *buf, uoffset_t end, uoffset_t base,
  */
 static inline int verify_vector(const void *buf, uoffset_t end, uoffset_t base, uoffset_t offset, uoffset_t elem_size, uint16_t align, uoffset_t max_count)
 {
-    uoffset_t n;
-
-    verify(check_aligned_header(end, base, offset, align), flatcc_verify_error_vector_header_out_of_range_or_unaligned);
+    verify(check_header(end, base, offset), flatcc_verify_error_vector_header_out_of_range_or_unaligned);
     base += offset;
-    n = read_uoffset(buf, base);
+
+    uoffset_t n = read_uoffset(buf, base);
     base += offset_size;
+
+#if FLATCC_TOLERATE_MISALIGNED_EMPTY_VECTORS
+    align = n == 0 ? uoffset_size : align;
+#endif
+    align = align < uoffset_size ? uoffset_size : align;
```

Comment:
You could place this in an #else branch. Then we have zero extra performance overhead.

Reply:
If `align = 1, n = 2` (for a vector of two bytes) then `align = n == 0 ? offset_size : align` will yield alignment 1. I think we need to keep the independent clamp to offset_size

Reply:
It doesn't matter because any alignment < offset_size will be overridden by the header check.

Reply:
hmm but would also be true in the other  case, so why is the line there?

We could remove it and it should be ok, but to be safe, we could also have:

verify(!(base & ((align - 1u) | (uoffset_size - 1u))